### ISSUE TITLE
cannot find module providing package github.com/Azure/azure-sdk-for-go/sdk/internal/runtime

### ISSUE THREAD

**Issue by adeturner** [Azure/azure-sdk-for-go#15384]
In WSL2, I'm naively trying to install the SDK, it cant find sdk/internal/runtime package

$ go version

go version go1.16.5 linux/amd64

$ go get -u -d github.com/Azure/azure-sdk-for-go/...

go: downloading github.com/Azure/azure-sdk-for-go v56.3.0+incompatible
go: downloading github.com/Azure/azure-sdk-for-go/sdk/armcore v0.8.0
go: downloading github.com/Azure/azure-sdk-for-go/sdk/azcore v0.19.0
go: downloading github.com/Azure/azure-sdk-for-go/sdk/internal v0.7.0
go: downloading github.com/Azure/azure-sdk-for-go/sdk/monitor/armmonitor v0.1.0
go: downloading github.com/Azure/go-autorest v14.2.0+incompatible
go: downloading github.com/Azure/go-autorest/autorest v0.11.19
go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0
go: downloading github.com/Azure/go-autorest/autorest/to v0.4.0
go: downloading github.com/Azure/go-autorest/autorest v0.11.20
go: downloading github.com/Azure/go-autorest/autorest/validation v0.3.1
go: downloading github.com/Azure/go-autorest/tracing v0.6.0
go: downloading github.com/Azure/go-autorest/logger v0.2.1
go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.13
go: downloading github.com/Azure/go-autorest/autorest/azure/auth v0.5.8
go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.15
go: downloading github.com/Azure/go-autorest/autorest/azure/cli v0.4.2
go: downloading github.com/Azure/go-autorest/autorest/azure/cli v0.4.3
go: downloading github.com/Azure/azure-sdk-for-go v0.2.0-beta
github.com/Azure/azure-sdk-for-go/sdk/armcore imports
        github.com/Azure/azure-sdk-for-go/sdk/internal/runtime: cannot find module providing package github.com/Azure/azure-sdk-for-go/sdk/internal/runtime

Just updating to mention I can see it in a branch only

[https://github.com/Azure/azure-sdk-for-go/blob/7d922f1ae7f000c2b6a7360503500a252fc962a2/sdk/internal/runtime/frame_error.go#L17]

It affects individual downloads of azcore too

**Comment by abelal83** [comment]
I'm facing this exact same issue today, various modules have this

```
PS C:\src\github\abx\demo> go get github.com/Azure/azure-sdk-for-go/sdk/cosmos/armcosmos
go: downloading github.com/Azure/azure-sdk-for-go/sdk/cosmos/armcosmos v0.1.0
github.com/Azure/azure-sdk-for-go/sdk/cosmos/armcosmos imports
        github.com/Azure/azure-sdk-for-go/sdk/armcore imports
        github.com/Azure/azure-sdk-for-go/sdk/internal/runtime: cannot find module providing package github.com/Azure/azure-sdk-for-go/sdk/internal/runtime
```

and

```
PS C:\src\github\abx\demo> go get github.com/Azure/azure-sdk-for-go/sdk/resources/armresources
github.com/Azure/azure-sdk-for-go/sdk/resources/armresources imports
        github.com/Azure/azure-sdk-for-go/sdk/armcore imports
        github.com/Azure/azure-sdk-for-go/sdk/internal/runtime: cannot find module providing package github.com/Azure/azure-sdk-for-go/sdk/internal/runtime
```

**Comment by jhendrixMSFT** [comment]
@abelal83 do you see this when creating a new app, or is there more to it?  I'm guessing the latter as it works for me with a new app, but maybe I'm missing something (my repro is with Go 1.17).
```
PS C:\git\jhendrixMSFT\armdemo> go mod init armdemo
go: creating new go.mod: module armdemo
PS C:\git\jhendrixMSFT\armdemo> go get github.com/Azure/azure-sdk-for-go/sdk/cosmos/armcosmos
go: downloading github.com/Azure/azure-sdk-for-go/sdk/cosmos/armcosmos v0.1.0
go get: added github.com/Azure/azure-sdk-for-go v56.1.0+incompatible
go get: added github.com/Azure/azure-sdk-for-go/sdk/armcore v0.8.0
go get: added github.com/Azure/azure-sdk-for-go/sdk/azcore v0.16.2
go get: added github.com/Azure/azure-sdk-for-go/sdk/cosmos/armcosmos v0.1.0
go get: added github.com/Azure/azure-sdk-for-go/sdk/internal v0.5.1
go get: added golang.org/x/net v0.0.0-20201010224723-4f7140c49acb
go get: added golang.org/x/text v0.3.3
PS C:\git\jhendrixMSFT\armdemo>
```

**Comment by abelal83** [comment]
Yup brand new app, 

```
go version go1.17 windows/amd64
```

I see the same result as you

```
PS C:\src\github\abelal83\test> go mod init armdemo
go: creating new go.mod: module armdemo
PS C:\src\github\abelal83\test> go get github.com/Azure/azure-sdk-for-go/sdk/cosmos/armcosmos
go: downloading github.com/Azure/azure-sdk-for-go v56.1.0+incompatible
go: downloading golang.org/x/net v0.0.0-20201010224723-4f7140c49acb
go get: added github.com/Azure/azure-sdk-for-go v56.1.0+incompatible
go get: added github.com/Azure/azure-sdk-for-go/sdk/armcore v0.8.0
go get: added github.com/Azure/azure-sdk-for-go/sdk/azcore v0.16.2
go get: added github.com/Azure/azure-sdk-for-go/sdk/cosmos/armcosmos v0.1.0
go get: added github.com/Azure/azure-sdk-for-go/sdk/internal v0.5.1        
go get: added golang.org/x/net v0.0.0-20201010224723-4f7140c49acb
go get: added golang.org/x/text v0.3.3
```

However, the issue arises when I use with other imports

```go
package main

import (
	"fmt"

	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
	"github.com/Azure/azure-sdk-for-go/sdk/cosmos/armcosmos" 
)

func main() {
	cred, err := azidentity.NewDefaultAzureCredential(nil)
	if err != nil {
		fmt.Print(err)
		return
	}
	fmt.Print(cred)
	con := arm.NewDefaultConnection(cred, nil)
	fmt.Print(con)

	client := armcosmos.NewDatabaseClient(con, "xyz")
}

```

The error I get in VS Code is
```
error while importing github.com/Azure/azure-sdk-for-go/sdk/cosmos/armcosmos: cannot find module providing package github.com/Azure/azure-sdk-for-go/sdk/internal/runtime: import lookup disabled by -mod=readonly
```

The other 2 packages import and work just fine.

**Comment by jhendrixMSFT** [comment]
Ah I see.  We're in the middle of refactoring the `azcore` module and the SDKs (e.g. `armcosmos`) haven't caught up yet.  This "catching up" will happen this week.  In the meantime, here's what you need to do to get unblocked.

- Use `azcore@v0.16.2`
- Use `armcore@v0.8.0` in lieu of `azcore/arm` (`armcore` has been merged into `azcore/arm` starting with `v0.19.0`)
- Use `azidentity@v0.9.2`

We'll also be updating our docs/samples as part of the refresh to reference the new packages.

**Comment by abelal83** [comment]
works perfectly now, thank you @jhendrixMSFT.

**Comment by adeturner** [comment]
Thanks

**Comment by adeturner** [comment]
Apologies, I had to reopen this; whilst the original issue is fixed (go get everything) the latest build is still problematic for me. Is there an easy workaround?  Starting with an empty cache...

sudo rm -rf ./pkg/mod/github.com/*azure* ./pkg/mod/cache/download/sumdb/sum.golang.org/lookup/github.com/*azure* ./pkg/mod/cache/download/github.com/*azure*

$ grep -i azure go.mod go.sum
<no results>

```
import (
	"github.com/Azure/azure-sdk-for-go/sdk/armservicebus"
	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
)

func NewAzServiceBusTest(sbNamespace, queueName string) (asb *AzServiceBus, err error) {
	cred, err := azidentity.NewDefaultAzureCredential(nil)
	con := arm.NewDefaultConnection(cred, nil)
	client := armservicebus.NewSubscriptionsClient(con, "<subscription ID>")
	fmt.Println(fmt.Sprintf("%v", client))
}
```

$ go mod tidy
go: finding module for package github.com/Azure/go-autorest/autorest
go: finding module for package github.com/Azure/azure-sdk-for-go/sdk/azcore/arm
go: finding module for package github.com/Azure/azure-sdk-for-go/sdk/azidentity
go: finding module for package github.com/Azure/azure-sdk-for-go/sdk/armservicebus
go: downloading github.com/Azure/azure-sdk-for-go v0.2.0-beta
go: downloading github.com/Azure/go-autorest v1.1.1
go: downloading github.com/Azure/azure-sdk-for-go/sdk/azidentity v0.11.0
go: downloading github.com/Azure/go-autorest/autorest v0.11.21
go: downloading github.com/Azure/azure-sdk-for-go/sdk/azcore v0.19.0
go: downloading github.com/Azure/azure-sdk-for-go v57.2.0+incompatible
go: downloading github.com/Azure/go-autorest v14.2.0+incompatible
go: finding module for package github.com/Azure/go-autorest/autorest/azure
go: finding module for package github.com/Azure/go-autorest/autorest/azure/auth
go: downloading github.com/Azure/go-autorest/autorest/azure/auth v0.5.8
go: found github.com/Azure/azure-sdk-for-go/sdk/azcore/arm in github.com/Azure/azure-sdk-for-go/sdk/azcore v0.19.0
go: found github.com/Azure/azure-sdk-for-go/sdk/azidentity in github.com/Azure/azure-sdk-for-go/sdk/azidentity v0.11.0
go: found github.com/Azure/go-autorest/autorest in github.com/Azure/go-autorest/autorest v0.11.21
go: found github.com/Azure/go-autorest/autorest/azure in github.com/Azure/go-autorest/autorest v0.11.21
go: found github.com/Azure/go-autorest/autorest/azure/auth in github.com/Azure/go-autorest/autorest/azure/auth v0.5.8
go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.14
go: downloading github.com/Azure/go-autorest/tracing v0.6.0
go: downloading github.com/Azure/go-autorest/logger v0.2.1
go: downloading github.com/Azure/go-autorest/autorest/mocks v0.4.1
go: downloading github.com/Azure/azure-sdk-for-go/sdk/internal v0.7.0
go: downloading github.com/Azure/go-autorest/autorest/azure/cli v0.4.2
go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0
go: finding module for package github.com/Azure/azure-sdk-for-go/sdk/armservicebus
github.com/Choreografy/consumerAPI/persistence/brokers imports
        github.com/Azure/azure-sdk-for-go/sdk/armservicebus: module github.com/Azure/azure-sdk-for-go@latest found (v57.2.0+incompatible), but does not contain package github.com/Azure/azure-sdk-for-go/sdk/armservicebus

**Comment by jhendrixMSFT** [comment]
I think the problem here is that your path to `armservicebus` is incorrect.  It should be `github.com/Azure/azure-sdk-for-go/sdk/servicebus/armservicebus`

See https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/servicebus/armservicebus

**Comment by adeturner** [comment]
Thanks Joel... I did get it working but forgot to update. I'll be honest, I didn't really understand how to use the servicebus library so gave up and wrapped az cli to get the job done, and then azure-service-bus-go for the client+topic/sub building. 

If there is any documentation or samples that explains how to use the auto-generated APIs with a local service principal I've love to see it.  That said, even for azure-service-bus-go I mostly had to read the code to work it out...

**Comment by sundowndev** [comment]
Hi, I have the same issue with package `github.com/Azure/azure-sdk-for-go/sdk/resources/armresources`.

```
$ go get github.com/Azure/azure-sdk-for-go/sdk/resources/armresources@v0.3.0 
go get: upgraded github.com/Azure/azure-sdk-for-go/sdk/azidentity v0.9.2 => v0.10.0
go get: added github.com/Azure/azure-sdk-for-go/sdk/resources/armresources v0.3.0

$ go run main.go
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/armcore@v0.8.0/policy_register_rp.go:19:2: no required module provides package github.com/Azure/azure-sdk-for-go/sdk/internal/runtime; to add it:
	go get github.com/Azure/azure-sdk-for-go/sdk/internal/runtime
```

I don't get why the error is about `github.com/!azure/azure-sdk-for-go/sdk/armcore` but it only happens when importing `armresources`. Is there a workaround for this ?

**Comment by eliecharra** [comment]
> github.com/!azure/azure-sdk-for-go/sdk/armcore

It seems that it worked by removing `armcore` and use `azcore/arm` instead. I also have to update `armstorage` to be able to pass `azcore.arm.Connection` to client constructor
